File: src/test/java/me/viniciusarnhold/altaria/command/PrefixesTest.java
package me.viniciusarnhold.altaria.command;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

/**
 * Created by Vinicius.
 *
 * @since ${PROJECT_VERSION}
 */
class PrefixesTest {

    @Test
    void getInstance() {
        assertNotNull(Prefixes.getInstance());
        assertSame(Prefixes.getInstance().getClass(), Prefixes.class);
    }

    @Test
    void current() {
        assertNotNull(Prefixes.getInstance().current());
        assertSame(Prefixes.getInstance().current(), "@");
    }

    @Test
    void changePrefix() {
        assertThrows(IllegalArgumentException.class, () -> Prefixes.getInstance().changePrefix(null));

        assertEquals(Prefixes.getInstance().changePrefix("!"), "!");
        assertEquals(Prefixes.getInstance().current(), "!");

        assertEquals(Prefixes.getInstance().changePrefix("0"), "0");
        assertEquals(Prefixes.getInstance().current(), "0");

    }

}

File: commands/src/test/kotlin/commands/PrefixesTest.kt
package me.viniciusarnhold.altaria.command

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

/**
 * Created by Vinicius.

 * @since ${PROJECT_VERSION}
 */
internal class PrefixesTest {

    @Test
    fun changePrefix() {
        var newPrefix = "!"
        assertThat(Prefixes.changePrefix(newPrefix)).isEqualTo(newPrefix)
        assertThat(Prefixes.current()).isEqualTo(newPrefix)

        newPrefix = "0"
        assertThat(Prefixes.changePrefix(newPrefix)).isEqualTo(newPrefix)
        assertThat(Prefixes.current()).isEqualTo(newPrefix)
    }
}